lib/tx_build/optional_claim_predicate.ex

```diff
@@ -18,13 +18,9 @@ defmodule Stellar.TxBuild.OptionalClaimPredicate do
     %__MODULE__{value: predicate}
   end
 
-  def new(nil, _opts), do: %__MODULE__{value: nil}
-
   def new(_args, _opts), do: {:error, :invalid_optional_claim_predicate}
 
   @impl true
-  def to_xdr(%__MODULE__{value: nil}), do: OptionalClaimPredicate.new()
```

Comment:
@einerzg I'm not really sure about this change, seems there are some things you're not taking into account here.

* **Optional** types will always require a nil value to represent **" the no presence of in the stellar network"** ... otherwise, the module should be renamed.
* The module is still setting nil as a default value at `new/2`  when the predicate is not passed through.
* The typespec for `predicate` is still setting `nil` as a valid data type. Actually, I'm not really sure if the typespec is needed.

Best

Reply:
yes you are right, I'll change it!!!